src/utils/string/StringUtils.js

```diff
@@ -385,8 +385,7 @@ const getDurationTimeText = (selectedtime) => {
 };
 
 const getCaptionWithLinkStyle = (text) => {
-  const linkRegex = /(https?:\/\/\S+)+/g;
-  return reactStringReplace(text, linkRegex, (match) => (
+  return reactStringReplace(text, urlRegex, (match) => (
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `onOpenLink` function is used here but it's not defined anywhere in this hunk. Make sure it's imported or defined in the same file.
```diff
-    return reactStringReplace(text, urlRegex, (match) => (
-      <HighlightText text={match} onPress={() => onOpenLink(match)} />
-    ));
+    return reactStringReplace(text, urlRegex, (match) => {
+      // Assuming onOpenLink is imported from somewhere
+      return <HighlightText text={match} onPress={() => onOpenLink(match)} />;
+    });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->